src/hooks/useActionCenterSubscriber/useLimitOrderActionSubscriber.tsx

```diff
@@ -0,0 +1,394 @@
+import { ExternalLinkIcon } from '@chakra-ui/icons'
+import {
+  Box,
+  Flex,
+  Icon,
+  Link,
+  Text as CText,
+  useColorModeValue,
+  usePrevious,
+  useToast,
+} from '@chakra-ui/react'
+import { OrderStatus } from '@shapeshiftoss/types'
+import { bnOrZero, fromBaseUnit } from '@shapeshiftoss/utils'
+import { useEffect, useMemo } from 'react'
+import { TbCircleCheckFilled, TbCircleXFilled } from 'react-icons/tb'
+import { useTranslate } from 'react-polyglot'
+import { v4 as uuidv4 } from 'uuid'
+
+import { useLocaleFormatter } from '../useLocaleFormatter/useLocaleFormatter'
+
+import { useLimitOrdersQuery } from '@/components/MultiHopTrade/components/LimitOrder/hooks/useLimitOrders'
+import { actionSlice } from '@/state/slices/actionSlice/actionSlice'
+import { ActionStatus, ActionType, isLimitOrderAction } from '@/state/slices/actionSlice/types'
+import { PriceDirection } from '@/state/slices/limitOrderInputSlice/constants'
+import {
+  selectBuyAmountCryptoBaseUnit,
+  selectInputSellAmountCryptoBaseUnit,
+} from '@/state/slices/limitOrderInputSlice/selectors'
+import { LimitOrderSubmissionState } from '@/state/slices/limitOrderSlice/constants'
+import { limitOrderSlice } from '@/state/slices/limitOrderSlice/limitOrderSlice'
+import {
+  selectActiveQuoteBuyAsset,
+  selectActiveQuoteExpirationTimestamp,
+  selectActiveQuoteId,
+  selectActiveQuoteLimitPrice,
+  selectActiveQuoteSellAsset,
+  selectLimitOrderSubmissionMetadata,
+} from '@/state/slices/limitOrderSlice/selectors'
+import {
+  selectLimitOrderActionByCowSwapQuoteId,
+  selectLimitOrderActionByLimitOrderId,
+  selectOpenLimitOrderActionsFilteredByWallet,
+} from '@/state/slices/selectors'
+import { store, useAppDispatch, useAppSelector, useSelectorWithArgs } from '@/state/store'
+
+type UseLimitOrderActionSubscriberProps = {
+  onDrawerOpen: () => void
+}
+
+export const useLimitOrderActionSubscriber = ({
+  onDrawerOpen,
+}: UseLimitOrderActionSubscriberProps) => {
+  const dispatch = useAppDispatch()
+  const translate = useTranslate()
+  const toastColor = useColorModeValue('white', 'gray.900')
+
+  const {
+    number: { toCrypto },
+  } = useLocaleFormatter()
+
+  const activeQuote = useAppSelector(limitOrderSlice.selectors.selectActiveQuote)
+  const sellAsset = useAppSelector(selectActiveQuoteSellAsset)
+  const buyAsset = useAppSelector(selectActiveQuoteBuyAsset)
+  const sellAmountCryptoBaseUnit = useAppSelector(selectInputSellAmountCryptoBaseUnit)
+  const buyAmountCryptoBaseUnit = useAppSelector(selectBuyAmountCryptoBaseUnit)
+  const quoteId = useAppSelector(selectActiveQuoteId)
```

Comment:
q: this will only work with the current limit order active quote, is this expected? Shouldn't this work similarly to `useSwapActionSubscriber` and iterate on open IDs?

Reply:
This is for the limit order initial insertion and update, then we rely on every orders from `useLimitOrders`

This is the exact same as we were doing for the swaps before we decided to make it another home for the sanity of the swap slice, see line 240, this effects relies on the orders instead of the quoteId

Reply:
Done in [2712581](https://github.com/shapeshift/web/pull/9626/commits/271258198517d54466a30d742e57de2d7e53f15b)